File: backend/models/Clase.js
const mongoose = require('mongoose');

const esquemaClase = new mongoose.Schema({
  nombre: {
    type: String,
    enum: ['funcional', 'pilates', 'zumba'],
    required: true
  },
  dia: {
    type: String,
    enum: ['Lunes', 'Martes', 'Miércoles', 'Jueves', 'Viernes'],
    required: true
  },
  horaInicio: {
    type: String, // Hora en formato "HH:mm"
    required: true
  },
  horaFin: {
    type: String,
    required: true
  },
  fecha: {
    type: Date, // Fecha exacta de la clase
    required: true
  },
  cuposDisponibles: { 
    type: Number, 
    default: 14 
  },
  maximoParticipantes: {
    type: Number,
    default: 14 // Máximo por defecto es 14, pero puede variar
  },
  listaEspera: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Usuario' // Usuarios en lista de espera
  }]
  // participantes y asistencias eliminados
}, { timestamps: true });

const Clase = mongoose.model('Clase', esquemaClase);

module.exports = Clase;

File: frontend/lib/models/clase.dart
class Clase {
  final String id;
  final String nombre;
  final String dia;
  final String horaInicio;
  final String horaFin;
  final DateTime fecha;
  final int cuposDisponibles;
  final int maximoParticipantes;
  final List<String> listaEspera;

  Clase({
    required this.id,
    required this.nombre,
    required this.dia,
    required this.horaInicio,
    required this.horaFin,
    required this.fecha,
    required this.cuposDisponibles,
    required this.maximoParticipantes,
    required this.listaEspera,
  });

  factory Clase.fromJson(Map<String, dynamic> json) {
    return Clase(
      id: json['_id'] ?? '',  // <- Usa _id porque mongo devuelve así
      nombre: json['nombre'] ?? '',
      dia: json['dia'] ?? '',
      horaInicio: json['horaInicio'] ?? '',
      horaFin: json['horaFin'] ?? '',
      fecha: DateTime.tryParse(json['fecha'] ?? '') ?? DateTime.now(),
      cuposDisponibles: json['cuposDisponibles'] ?? 0,
      maximoParticipantes: json['maximoParticipantes'] ?? 0,
      listaEspera: (json['listaEspera'] as List?)?.map((e) => e.toString()).toList() ?? [],
    );
  }
}
